### ISSUE TITLE
java.lang.RuntimeException: Unknown exception code: 1 msg null

### ISSUE THREAD

**Issue by rfgamaral** [Karumi/Dexter#86]
I don't have that many users but I have a few occurrences of this exception. Not many for the number of users but still it's annoying that it crashes for some people.

```
Caused by java.lang.RuntimeException: Unknown exception code: 1 msg null
       at android.os.Parcel.readException(Parcel.java:1477)
       at android.os.Parcel.readException(Parcel.java:1419)
       at android.app.ActivityManagerProxy.checkPermission(ActivityManagerNative.java:3518)
       at android.app.ContextImpl.checkPermission(ContextImpl.java:1747)
       at android.content.ContextWrapper.checkPermission(ContextWrapper.java:545)
       at android.support.v4.content.ContextCompat.checkSelfPermission(SourceFile:425)
       at com.karumi.dexter.AndroidPermissionService.checkSelfPermission(SourceFile:34)
       at com.karumi.dexter.DexterInstance.getPermissionStates(SourceFile:190)
       at com.karumi.dexter.DexterInstance.onActivityReady(SourceFile:127)
       at com.karumi.dexter.Dexter.onActivityReady(SourceFile:162)
       at com.karumi.dexter.DexterActivity.onCreate(SourceFile:31)
       at android.app.Activity.performCreate(Activity.java:5318)
       at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1088)
       at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2302)
       at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2390)
       at android.app.ActivityThread.access$800(ActivityThread.java:151)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1321)
       at android.os.Handler.dispatchMessage(Handler.java:110)
       at android.os.Looper.loop(Looper.java:193)
       at android.app.ActivityThread.main(ActivityThread.java:5292)
       at java.lang.reflect.Method.invokeNative(Method.java)
       at java.lang.reflect.Method.invoke(Method.java:515)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:825)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:641)
       at dalvik.system.NativeStart.main(NativeStart.java)
```

AFAIK, this is the same issue as hotchemi/PermissionsDispatcher#107 and Appboy/appboy-android-sdk#42.

The solution has been to wrap `android.support.v4.content.ContextCompat.checkSelfPermission` in a try/catch block.

I have a PR for this in this works (credits to original fix developers) which I'll post in a few minutes...

**Comment by Serchinastico** [comment]
Hi @rfgamaral 

Very interesting indeed, I wasn't aware of the issue TBH. 
Your PR will be more than welcome. We will take care of doing a release as soon as we review it and merge it so that you can include it in your project.

Thank you very much for reporting!

**Comment by rfgamaral** [comment]
Just made the PR, waiting for the build to pass...

> We will take care of doing a release as soon as we review it and merge it so that you can include it in your project.

Great, thank you very much :)

**Comment by Serchinastico** [comment]
This issue has been fixed in Dexter 2.3.1 thanks to @rfgamaral (thanks, buddy! 👌)
